evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -165,14 +168,18 @@ def transform(self, X, y=None):
             X_not_all_null[X_interpolate.columns] = imputed
 
         y_imputed = pd.Series(y)
-        if self._impute_target == "forwards_fill":
-            y_imputed = y.pad()
-            y_imputed.bfill(inplace=True)
-        elif self._impute_target == "backwards_fill":
-            y_imputed = y.bfill()
-            y_imputed.pad(inplace=True)
-        elif self._impute_target == "interpolate":
-            y_imputed = y.interpolate()
-            y_imputed.bfill(inplace=True)
+        if y is not None and len(y) > 0:
```

Comment:
There was a test case where the passed in y was an empty Series instead of being None. Alternative would have been to change the test, I figured this was more robust but happy to swap.